### DISCUSSION TITLE
Avoid generating CSS for irrelevant content

### DISCUSSION THREAD

**Discussion by hungtcs** [tailwindlabs/tailwindcss#12192] [Help] [answered]
Hello everyone,

I found that keywords contained in strings that have nothing to do with the DOM will also generate css. 
Is there a way to avoid this?

<img width="1429" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/11692935/fe4c4456-2aa7-4ea3-9fb5-c0704dbfd037">

**Comment by alex-krasikau** [comment, marked as answer]
Hello, @hungtcs!

1. Tailwind is a technology-agnostic framework that works with anything you might want to use by parsing plain-text files for keywords. Making it smarter would make it less flexible, require adjustments for every framework where it is used, and require a much more complex parser, which doesn't make much sense.

2. Even using hundreds of utilities in the copy for your website will not have a significant impact on your website's performance. However, if you want to avoid any impact at all costs, you can consider moving your content into a separate folder and excluding this folder from the content settings in your Tailwind Config.

**Reply by hungtcs** [reply]
Thanks for your answer,  sometimes I really have obsessive-compulsive disorder haha.
